File: 12-2025-2026/_feladat/backend/feladat_2026_01_20_oroklodes/demo_nyilvantartas/src/Neu/Nyilvantartas/Szemely.php
<?php

namespace Neu\Nyilvantartas;

use DateTime;
use LogicException;
use Stringable;

class Szemely implements Stringable {
    protected string $nev;
    protected DateTime $szuletett;

    public function __construct(string $nev, DateTime $szuletett) {
        $this->nev = $nev;
        $this->szuletett = $szuletett;
    }

    public function __get(string $tulajdonsag): mixed {
        if (in_array($tulajdonsag, array_keys(get_object_vars($this)))) {
            return $this->$tulajdonsag;
        }

        throw new LogicException("Nem olvasható tulajdonság: $tulajdonsag");
    }

    public function __tostring(): string {
        return $this->nev . " " . $this->szuletett->format("Y-m-d");
    }
}

File: 12-2025-2026/_feladat/alkalmazasfejlesztes/feladat_2026_01_20_halmaz/Halmazok_VB/Halmazok_VB_Lib/Kerdoiv/Szemely.cs
namespace Halmazok_VB_Lib.Kerdoiv;

public abstract class Szemely
{
    public string OktatsiAzonosito { get; set; }
    public string Nev { get; set; }

    protected Szemely(string oktatsiAzonosito, string nev)
    {
        OktatsiAzonosito = oktatsiAzonosito;
        Nev = nev;
    }

    public override string ToString()
    {
        return $"{OktatsiAzonosito} - {Nev}";
    }
}